OpenRA.Mods.Common/Traits/Render/WithShadow.cs

```diff
@@ -16,31 +16,44 @@
 
 namespace OpenRA.Mods.Common.Traits.Render
 {
-	[Desc("Clones the aircraft sprite with another palette below it.")]
-	class WithShadowInfo : ITraitInfo
+	[Desc("Clones the actor sprite with another palette below it.")]
+	public class WithShadowInfo : UpgradableTraitInfo
 	{
 		[PaletteReference] public readonly string Palette = "shadow";
 
-		public object Create(ActorInitializer init) { return new WithShadow(this); }
+		[Desc("Shadow position offset relative to actor position (ground level).")]
+		public readonly WVec Offset = WVec.Zero;
+
+		[Desc("Shadow Z offset relative to actor sprite.")]
+		public readonly int ZOffset = -5;
```

Comment:
> Hm, couldn't we just include this in the normal Offset?

`Offset` is a world position offset, while `ZOffset` could also be called `ZSortingOffset`, but we already call it just `ZOffset` everywhere else.